src/librbd/cache/pwl/LogEntry.cc

```diff
@@ -71,18 +71,18 @@ std::ostream &operator<<(std::ostream &os,
 void WriteLogEntry::init(bool has_data,
                          uint64_t current_sync_gen,
                          uint64_t last_op_sequence_num, bool persist_on_flush) {
-  ram_entry.has_data = 1;
+  ram_entry.set_data_stored(has_data);
```

Comment:
We weren't looking at `has_data` parameter here previously, was that a bug?

Reply:
Yes, but it won't cause error. Only one func call it and pass `true`.
`C_WriteRequest<T>::setup_log_operations()` => `operation->init(true, ...)` => `WriteLogOperation::init() `=> `WriteLogEntry::init()`